pkg/nodemanager/nodemanager.go

```diff
@@ -602,19 +598,33 @@ func nodeAddressesChangeDetected(addressSet1, addressSet2 []v1.NodeAddress) bool
 	return false
 }
 
-func ensureNodeProvidedIPExists(node *v1.Node, nodeAddresses []v1.NodeAddress) (*v1.NodeAddress, bool) {
-	var nodeIP *v1.NodeAddress
-	nodeIPExists := false
-	if providedIP, ok := node.ObjectMeta.Annotations[cloudproviderapi.AnnotationAlphaProvidedIPAddr]; ok {
-		nodeIPExists = true
-		for i := range nodeAddresses {
+// Ensure all provided node ip addresses are found in the cloud provider, otherwise return false
+// and the existing node addresses.
+// When there's no provided node ip addresses, it will return true.
+func ensureNodeProvidedIPsExists(node *v1.Node, nodeAddresses []v1.NodeAddress) ([]v1.NodeAddress, bool) {
+	providedIPStr, ok := node.ObjectMeta.Annotations[cloudproviderapi.AnnotationAlphaProvidedIPAddr]
+	if !ok || len(providedIPStr) == 0 {
+		return []v1.NodeAddress{}, true
+	}
+	var existNodeIPs []v1.NodeAddress
+	nodeIPsExists := false
+	providedIPs := strings.Split(providedIPStr, ",")
+	for i := range nodeAddresses {
+		for _, providedIP := range providedIPs {
+			fmt.Println(nodeAddresses[i].Address, providedIP)
+
```

Comment:
could you trim spaces of providedIP before compare?

Reply:
Good point, I detected this in the UT as well. wanted to ask whether it's necessary or not.
Included an UT to test extra space as well.